diff -rupN org/acmsl/queryj/templates/packaging/PerForeignKeyTemplatesTest.stg org/acmsl/queryj/templates/packaging/PerForeignKeyTemplatesTest.stg
--- org/acmsl/queryj/templates/packaging/PerForeignKeyTemplatesTest.stg
+++ org/acmsl/queryj/templates/packaging/PerForeignKeyTemplatesTest.stg
@@ -81,6 +81,7 @@ import org.acmsl.queryj.metadata.engines.oracle.OracleEngine;
 import org.acmsl.queryj.metadata.SqlDAO;
 import org.acmsl.queryj.metadata.vo.Attribute;
 import org.acmsl.queryj.metadata.vo.ForeignKey;
+import org.acmsl.queryj.metadata.vo.Row;
 import org.acmsl.queryj.metadata.vo.Table;
 import org.acmsl.queryj.QueryJCommand;
 import org.acmsl.queryj.QueryJCommandWrapper;
@@ -370,7 +371,10 @@ public class PerForeignKeyTemplatesTest
 
             @NotNull final MetadataManager t_MetadataManager =
                 retrieveMetadataManager(
-                    engine, new ArrayList\<>(tables.values()), new ArrayList\<>(0), decoratorFactory);
+                    engine,
+                    new ArrayList\<Table\<String, Attribute\<String>, List\<Attribute\<String\>>\>>(tables.values()),
+                    new ArrayList\<Row\<String\>>(0),
+                    decoratorFactory);
 
             new QueryJCommandWrapper<MetadataManager>(t_Command).setSetting(
                 DatabaseMetaDataRetrievalHandler.METADATA_MANAGER, t_MetadataManager);
