diff -rupN org/acmsl/queryj/metadata/TableAttributesListDecoratorTest.java org/acmsl/queryj/metadata/TableAttributesListDecoratorTest.java
--- org/acmsl/queryj/metadata/TableAttributesListDecoratorTest.java
+++ org/acmsl/queryj/metadata/TableAttributesListDecoratorTest.java
@@ -41,6 +41,7 @@ import org.acmsl.queryj.customsql.CustomSqlProvider;
 import org.acmsl.queryj.metadata.engines.JdbcMetadataTypeManager;
 import org.acmsl.queryj.metadata.vo.AbstractAttribute;
 import org.acmsl.queryj.metadata.vo.Attribute;
+import org.acmsl.queryj.metadata.vo.ForeignKey;
 import org.acmsl.queryj.metadata.vo.Table;
 import org.acmsl.queryj.metadata.vo.TableValueObject;
 
@@ -98,9 +99,9 @@ public class TableAttributesListDecoratorTest
             new TableValueObject(
                 "name",
                 "comment",
-                new ArrayList<>(0),
-                new ArrayList<>(0),
-                new ArrayList<>(0),
+                new ArrayList<Attribute<String>>(0),
+                new ArrayList<Attribute<String>>(0),
+                new ArrayList<ForeignKey<String>>(0),
                 null, // parent
                 null, // static
                 false, // vo decorated
